server.js

```diff
@@ -40,14 +43,74 @@ app.use('/api', router);
 
 router.get('/feedItems', function (req, res) {
     console.log('GET Feed Items');
-    // get all the feedItems
-    FeedItem.find({}, function(err, feedItems) {
-        if (err) throw err;
-
-        // array of feed items
-        console.log(feedItems);
-        res.json(feedItems);
-    });
+    var requestedCount = parseInt(req.query.count);
+    var requestedIndex = parseInt(req.query.index);
+    var requestedType = req.query.type;
+    console.log('Requested Count: ' + requestedCount);
+    console.log('Requested Index: ' + requestedIndex);
+    console.log('Requested Type: ' + requestedType);
+
+    if (requestedType && requestedCount && requestedIndex) {
+        FeedItem.find({type: requestedType, index: {$gt: requestedIndex}})
+            .limit(requestedCount)
+            .sort('index')
```

Comment:
Does this sort asc or desc by default? I would think biggest index == newest item == top of the list, right? So we want descending?